src/main/scala/mesosphere/marathon/api/CORSFilter.scala

```diff
@@ -0,0 +1,35 @@
+package mesosphere.marathon.api
+
+import javax.servlet._
+import javax.servlet.http.{ HttpServletRequest, HttpServletResponse }
+
+import scala.collection.JavaConverters._
+
+class CORSFilter extends Filter {
+  override def init(filterConfig: FilterConfig): Unit = {}
+
+  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
+
+    response match {
+      case httpResponse: HttpServletResponse =>
```

Comment:
we should completely disable this by default, so assuming we have a flag --enable_cors we should add a check here:

`case httpResponse: HttpServletResponse if config.enableCORS =>`